spec/generator.json

```diff
@@ -31,10 +31,19 @@
             ]
         },
 
+        "file": {
+            "description": "Represents a source file",
+            "fields": [
+                { "name": "name", "type": "string", "description": "The recommended name for the file, including the path to the file, where appropriate.", "example": "com/myorg/myapplication/Foo.java" },
```

Comment:
My original pull request had refactored the generators to always return a list of files and then the invocations controller was concatenating them into the 'source' field for backwards compatibility. I think we still need that change, right? I can make it as part of the java_models pull request if you want.